ddtrace/internal/compat.py

```diff
@@ -242,3 +244,10 @@ def ensure_pep562(module_name):
     def ensure_pep562(module_name):
         # type: (str) -> None
         pass
+
+
+def maybe_stringify(obj):
```

Comment:
I don't think this is wroth a function and even less in `compat`.

Reply:
That's here for future use ;). Maybe not useful now but might turn out to be a pattern that needs to be used elsewhere too

Reply:
I wonder if this is logic that should live inside of `_set_str_tag`, or `set_tag` (and we dedicate ourselves to making `set_tag` more performant, etc)

Reply:
> I wonder if this is logic that should live inside of `_set_str_tag`, or `set_tag` (and we dedicate ourselves to making `set_tag` more performant, etc)

I have thought about it; I don't think it would be wise to put this logic in `_set_str_tag` as I believe this method should have a "strict" signature and only accept strings. It might make sense to put it in `set_tag` since the signature is more "lax".

Reply:
@majorgreys and I were thinking about this a bit. Because Python is Python (and we can't get type hints from all our integrations) I think it might make more sense to have `_set_str_tag` handle `None`.

Without type checking it's all too easy to forget to verify the types to `_set_str_tag` and we've been bitten by it here and in #2445 and #2298. Another solution might be that we test more rigorously but given each of the occurrences so far I don't think it's feasible that we have the amount of test coverage to handle all these cases.

Reply:
> I think it might make more sense to have `_set_str_tag` handle `None`.

This makes a lot of sense, however it won't be enough in cases like these since not only we need to handle `None`, but also convert to the expected type (`str` in this case) before calling the `None`-safe method. The conversion itself needs to take care of `None` for in this case, if we don't catch that, we would get the literal string `"None"` back, which we would then hand over to the method in place of the actual `None`.

Reply:
hmm yeah. What if we did something like

```python
def _set_str_tag(self, k, v):
  if isinstance(v, six.text_types):
    self.meta[k] = six.ensure_text(v)
  else:
    log.warning("Expected string type ...")
```

or maybe we just want to do a `try..except` around the ensure_text with a warning (I think this is the direction @majorgreys and I were leaning toward earlier and is similar to what we do with `set_tag`. It sucks but I'm not sure we can do better with Python being Python 😢

Reply:
Makes sense to me, bu the log statement concerns me. Not only because it might spam the logs, but also because it's not the behaviour I'd expect from a function that expects a certain type but gets something of a different type. So if we stipulate that `_set_str_tag` accepts a string and we pass something else, this should probably raise a `TypeError`. WDYT?

Reply:
I think I'm also in favour of allowing `_set_str_tag` to handle None types as mentioned by @Kyle-Verhoog, but to also raise a TypeError if it gets a `None` input. I'm open to this PR as is (LGTM), but just my personal thoughts...

Reply:
Is thet a chance to use the new `DD_TESTING_RAISE` to raise on testing at least, to catch easy errors?

Reply:
That was done by #2502